fedbiomed/researcher/federated_workflows/_experiment.py

```diff
@@ -798,7 +798,7 @@ def run_once(self, increase: bool = False, test_after: bool = False) -> int:
         optim_aux_var = self._collect_optim_aux_var()
 
         # update node states when used node list has changed from one round to another
-        self._update_nodes_states_agent(before_training=True)
+        self._update_nodes_states_agent(training_nodes, before_training=True)
```

Comment:
we might want to use the nodes specified dataset (`self._data.data()`) because you may sample some nodes for this round, but you want to consider here all Nodes available for training (`training_nodes` is a subset of `self._data.data()`) . While this issue doesnot happen with the `default_strategy`, it may be happening with strategies that involve sampling Nodes, such as random sampling